src/Error/Middleware/ErrorHandlerMiddleware.php

```diff
@@ -114,18 +117,32 @@ public function handleException($exception, $request, $response)
             $this->logException($request, $exception);
 
             return $res;
-        } catch (Exception $e) {
-            $this->logException($request, $e);
-
-            $body = $response->getBody();
-            $body->write('An Internal Server Error Occurred');
-            $response = $response->withStatus(500)
-                ->withBody($body);
+        } catch (Throwable $exception) {
+            $this->logException($request, $exception);
+            $response = $this->handleInternalError($response);
+        } catch (Exception $exception) {
+            $this->logException($request, $exception);
+            $response = $this->handleInternalError($response);
         }
 
         return $response;
     }
 
+    /**
+     * @param \Psr\Http\Message\ResponseInterface $response The response
+     *
+     * @return \Psr\Http\Message\ResponseInterface A response
+     */
+    protected function handleInternalError($response)
+    {
+        $body = $response->getBody();
+        $body->write('An Internal Server Error Occurred');
+        $response = $response->withStatus(500)
```

Comment:
Nitpick: Could directly return here instead of assigning var.

Reply:
Sometimes it is just easier to debug immutable objects with a debugger when you can set a break point before returning. But I will change it then.